### PR TITLE
Drone Ship Vault

### PR DESCRIPTION
**Pull request by JMWTurner**
drone ship as in, the ship is a drone, not that it contains any drones. only harmless mannequins. :^)

droneship

:cl:
* Added a new vault: the Drone Ship.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
What's in that room behind the blast doors?

**Comment by Intigracy** [comment]
```
aaaaaaaaaaabccdaaa
aaaaeccfcccccccdaa
gahaijklmcnopqrccd
asaatuvwxyxxxxxzAc <-- A is the stuff, c is the wall
aaBaiCDlEcnFGHnccI
aaaaJccfcccccccIaa
aaaaaaaaaaabccIaaa
```

"A" = (/obj/item/weapon/gun/osipr{pixel_y = 8},/obj/item/energy_magazine/osipr{pixel_x = 8},/obj/item/energy_magazine/osipr{pixel_x = 8; pixel_y = -6},/obj/item/osipr_core{pixel_y = -4},/obj/item/osipr_core{pixel_x = -4; pixel_y = -4},/obj/item/osipr_core{pixel_x = -8; pixel_y = -4},/turf/simulated/floor/greengrid{light_color = "#A9F291"; light_range = 2},/area/vault/droneship)

**Comment by JMWTurner** [comment]
An OSIPR, along with two rechargeable magazines and 3 dark energy cores. A reward to match the difficulty to obtain it.

**Comment by Sprok0** [comment]
I don't really know if the osipr should be obtainable in-game outside of bus/summon guns, it's a 50 brute damage gun that also stuns and destroys walls, with a 30 shot rechargeable magazine. It's a ballistic version of the pulse rifle.

**Comment by JMWTurner** [comment]
That's why I said a reward to match the difficulty to obtain it. The pod door's id_tag is different every round, so you need to kill the mannequins to get the code. Cyber mannequins have carp speed, 5 times as much HP, and they're all equipped with weapons that deal at least 15 damage, the one behind the glass who holds the code deals 40 damage. If you're dangerous enough to get rid of those guys successfully, then you don't need the OSIPR to be a danger to the station.

**Comment by Sprok0** [comment]
Actually the RPD thing I was thinking of doesn't work on shuttle floor turfs

**Comment by JMWTurner** [comment]
You're right. I realized that before you even answered, so here we go:

revision 1
- The droneship's area is now super jammed, not even telecrystals can help the telepad get through here.
- Trapped Cyber Mannequins don't awaken until you break their glass shield

What was that about an RPD though? You can't deconstruct shuttle walls can you?

File: maps/randomvaults/objects.dm [hunk 1]
```diff
@@ -104,6 +104,45 @@
...
+	requires_power = 1
+	jammed = 2
+	var/pod_code = "00000"
+
+/area/vault/droneship/New()
+	..()
+	pod_code = "[rand(10000, 99999.0)]"
+
+/obj/machinery/door/poddoor/droneship
+	name = "OSIPR Pod-Door"
```
Comment (PJB3005) [inline maps/randomvaults/objects.dm]: Should have improper

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: maps/randomvaults/objects.dm (hunk 1 above)
Comment (JMWTurner) [reply maps/randomvaults/objects.dm]: done

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**